usrp_drivers/usrp_driver.cpp

```diff
@@ -366,7 +373,15 @@ void transmit(zmq::context_t &driver_c, USRP &usrp_d, const DriverOptions &drive
       }() // full usrp function lambda
     ); // full usrp function timeit macro
 
+    rxsamplesmetadata::RxSamplesMetadata samples_metadata;
+
+    system_time = std::chrono::system_clock::now();
+    system_since_epoch = std::chrono::duration<double>(system_time.time_since_epoch());
+    // get_real_secs() may lose precision of the fractional seconds, but it's close enough
+    gps_to_system_time_diff = system_since_epoch.count() - box_time.get_real_secs();
```

Comment:
Why discard the fractional part? Your time duration has nanosecond precision.

Reply:
oh get_real_secs keeps both the whole and fractional seconds portions. It's so I can get a double to subtract from system time, that comment is from the ettus documentation here: https://files.ettus.com/manual/classuhd_1_1time__spec__t.html#a6ba38d752eeb3de17a3a3589a0b0decf

Reply:
sounds good